.github/workflows/worker-deploy.yml

```diff
@@ -0,0 +1,51 @@
+name: Deploy Worker
+
+on:
+  push:
+  workflow_dispatch:
+
+jobs:
+  deploy-to-cloudflare:
+    runs-on: ubuntu-latest
+    name: Automatic Cloudflare Deploy
+    steps:
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20.10.0"
+
+      - name: Enable corepack
+        run: corepack enable
+
+      - uses: actions/checkout@v4
+
+      - name: Update wrangler.toml Name Field
+        run: |
+          branch_name=$(echo '${{ github.event.ref }}' | sed 's#refs/heads/##' | sed 's#[^a-zA-Z0-9]#-#g')
+          # Extract base name from wrangler.toml
+          base_name=$(grep '^name = ' wrangler.toml | sed 's/^name = "\(.*\)"$/\1/')
+          # Concatenate branch name with base name
+          new_name="${base_name}-${branch_name}"
+          # Truncate the new name to 63 characters for RFC 1035
+          new_name=$(echo "$new_name" | cut -c 1-63)
+          # Update the wrangler.toml file
+          sed -i "s/^name = .*/name = \"$new_name\"/" wrangler.toml
+          echo "Updated wrangler.toml name to: $new_name"
+      - name: Deploy with Wrangler
+        id: wrangler_deploy
+        uses: cloudflare/wrangler-action@v3
+        with:
+          wranglerVersion: "3.61.0"
+          apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
+          accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
+          secrets: |
+            SUPABASE_URL
+            SUPABASE_KEY
+        env:
+          SUPABASE_URL: ${{ secrets.SUPABASE_URL }}
```

Comment:
I thought we got rid of the database dependency

Reply:
The wallet seems to be used to retrieve the wallet of the user.